docs/web/contribute/backend_test_suite.md

```diff
@@ -0,0 +1,115 @@
+# Working with the Backend Test Suite
+
+!!! danger "Before you start"
+
+    This section assumes you've already [set up a development environment](environment.md).
+
+!!! info "You may be able to skip this section"
+
+    If you haven't made changes to any core parts of ibis (e.g., `ibis/expr`)
+    or any specific backends (`ibis/backends`) this material isn't necessary to
+    follow to make a pull request.
+
+One the primary challenges when working with the ibis codebase is testing.
+
+Ibis supports a execution against a number of backends with wildly varying
+deployment models:
+
+- Single-node systems like SQLite
+- Multi-node client-server systems like PostgreSQL and MySQL
+- Systems that span a variety of different models like ClickHouse
+- Systems that are designed to run on premises, like Impala
+
+This presents quite a challenge for testing. This page is all about how to work
+with the backend test suite.
+
+## Systems that can be tested without any additional infrastructure
+
+Many backends can be tested without without `docker compose`.
+
+The SQLite, datafusion, and any pandas-based backends can all be tested without
+needing to do anything except loading data.
+
+## Backend Testing with Compose
+
+Here is the list of backends that can be tested using docker compose.
+
+| Backend    | Docker Compose Services |
+| ---------- | ----------------------- |
+| ClickHouse | `clickhouse`            |
+| PostgreSQL | `ibis-postgres`         |
+| impala     | `impala`                |
+| kudu       | `kudu`, `impala`        |
+| mysql      | `mysql`                 |
+
+### Testing a Compose Service
+
+!!! tip
+
+    Make sure you're inside of your clone of the ibis GitHub repository
+
+Let's fire up a PostgreSQL server and run tests against it.
+
+#### Start the Service
+
+Open a new shell and run
+
+```sh
+docker compose up ibis-postgres
```

Comment:
```suggestion
docker-compose up ibis-postgres
```

Reply:
Hm, this _should_ work even with older `docker-compose`. Does it not work for you?

Reply:
```shell
$ docker compose up ibis-postgres
docker: 'compose' is not a docker command.
See 'docker --help'
```
and that's on a pretty recent `docker`

```
$ docker version
Client:
 Version:           20.10.12
 API version:       1.41
 Go version:        go1.17.5
 Git commit:        e91ed5707e
 Built:             Mon Dec 13 22:31:40 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.12
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.5
  Git commit:       459d0dfbbb
  Built:            Mon Dec 13 22:30:43 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.5.9
  GitCommit:        1407cab509ff0d96baa4f0eb6ff9980270e6e620.m
 runc:
  Version:          1.0.3
  GitCommit:        v1.0.3-0-gf46b6ba2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
  ```

Reply:
ahh, right, it's the `docker-compose` version, innit?

Reply:
ok, comment retracted

Reply:
I think I'm going to move us back to 1, because most folks will probably have that for a while. v2 will probably work, though no guarantees :)

Reply:
Fixed.